examples/cloudflare-worker-multi-page/src/index.ts

```diff
@@ -0,0 +1,171 @@
+  // MY_DURABLE_OBJECT: DurableObjectNamespace
+  //
+  // Example binding to R2. Learn more at https://developers.cloudflare.com/workers/runtime-apis/r2/
+  // MY_BUCKET: R2Bucket
+}
+
+function parsePreviousFrameState(
+  searchParams: URLSearchParams,
+  initialState: any
+) {
+  try {
+    const stateValue = searchParams.get("s");
+
+    if (!stateValue) {
+      return initialState;
+    }
+
+    const parsedValue = JSON.parse(stateValue);
+
+    return parsedValue && typeof parsedValue === "object"
+      ? parsedValue
+      : initialState;
+  } catch (e) {
+    return initialState;
+  }
+}
+
+export default {
+  async fetch(req: Request, env: Env, ctx: ExecutionContext) {
+    const url = new URL(req.url);
+    const totalPages = 5;
+    const state = parsePreviousFrameState(url.searchParams, { pageIndex: 0 });
+
+    if (req.method === "POST") {
+      const body = await req.json();
+      const untrustedMessage = getFrameMessageFromRequestBody(body);
+
+      if (!untrustedMessage.data?.frameActionBody) {
+        return new Response("Invalid message", { status: 400 });
+      }
+
+      /* const result = await validateFrameMessage(body, {
+        hubHttpUrl: "https://nemes.farcaster.xyz:2281",
+        hubRequestOptions: {},
+      });
+      console.log(result);
+
+      if (!result.isValid) {
+        return new Response("Invalid message", { status: 400 });
+      }*/
+
+      const buttonIndex = untrustedMessage.data?.frameActionBody?.buttonIndex;
+
+      state.pageIndex = buttonIndex
+        ? Math.max(
+            0,
+            (state.pageIndex + (buttonIndex === 2 ? 1 : -1)) % totalPages
+          )
+        : state.pageIndex;
+    }
+
+    const imgResponse = new ImageResponse(
+      {
+        type: "div",
+        key: "1",
+        props: {
+          style: {
+            display: "flex", // Use flex layout
+            flexDirection: "row", // Align items horizontally
+            alignItems: "stretch", // Stretch items to fill the container height
+            width: "100%",
+            height: "100vh", // Full viewport height
+            backgroundColor: "white",
+          },
+          children: {
+            type: "div",
+            props: {
+              style: {
+                display: "flex",
+                flexDirection: "column",
+                justifyContent: "center",
+                alignItems: "center",
+                lineHeight: 1.2,
+                fontSize: 36,
+                color: "black",
+                flex: 1,
+                overflow: "hidden",
+              },
+              children: {
+                type: "div",
+                props: {
+                  tw: "flex flex-col",
+                  children: [
+                    {
+                      type: "img",
+                      props: {
+                        alt: "Image",
+                        src: `https://picsum.photos/seed/frames.js-${state.pageIndex}/1146/600`,
```

Comment:
how slow is fetching this image? can we provide images locally instead in a faster way?

Reply:
This isn't that slow, I tried repaately and got like 70ms for whole handler. We are losing time in loading the function to edge memory because it is 2.7MB.

Reply:
hm ok so the image generation just doesn't work with cloudflare workers. lets remove it